File: 프로그래머스/0/120895. 인덱스 바꾸기/인덱스 바꾸기.java
class Solution {
    public String solution(String my_string, int num1, int num2) {
        String answer = "";
        char x = my_string.charAt(num1);
        char y = my_string.charAt(num2);
        
        for(int i=0; i<my_string.length(); i++){
            if(i==num1) answer+= y;
            else if(i==num2) answer+= x;
            else answer+=my_string.charAt(i);
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120895. 인덱스 바꾸기/인덱스 바꾸기.kt
class Solution {
    fun solution(my_string: String, num1: Int, num2: Int): String {
        val sb = StringBuilder()
        my_string.mapIndexed{ i, v ->
            if(i == num1) sb.append(my_string.get(num2))
            else if(i == num2) sb.append(my_string.get(num1))
            else sb.append(v)
        }
        
        return sb.toString()
    }
}